### PR TITLE
Add Thymeleaf templating feature

### PR DESCRIPTION
**Pull request by t-kameyama**
Fixes https://github.com/ktorio/ktor/issues/709

### CODE REVIEW

File: ktor-features/ktor-thymeleaf/jvm/src/io/ktor/thymeleaf/Thymeleaf.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    val template: String,
+    val model: Map<String, Any>,
+    val etag: String? = null,
+    val contentType: ContentType = ContentType.Text.Html.withCharset(Charsets.UTF_8)
+)
+
+/**
+ * Thymeleaf support feature. Provides ability to respond with [Thymeleaf]
+ */
+class Thymeleaf(private val engine: TemplateEngine) {
```
Comment (cy6erGn0m) [inline ktor-features/ktor-thymeleaf/jvm/src/io/ktor/thymeleaf/Thymeleaf.kt]: Doesn't it require/provide any configuration? For example, resolution rules?

Comment (t-kameyama) [reply ktor-features/ktor-thymeleaf/jvm/src/io/ktor/thymeleaf/Thymeleaf.kt]: https://github.com/t-kameyama/ktor/blob/b726a79ed877d57ebe970383dde42eb7cd81d05b/ktor-features/ktor-thymeleaf/jvm/test/it/ktor/thymeleaf/ThymeleafTest.kt#L142-L148

Comment (t-kameyama) [reply ktor-features/ktor-thymeleaf/jvm/src/io/ktor/thymeleaf/Thymeleaf.kt]: We set configuration like this:
```kotlin
application.install(Thymeleaf) {
    val resolver = ClassLoaderTemplateResolver()
    resolver.setTemplateMode("HTML")
    resolver.prefix = "templates/"
    resolver.suffix = ".html"
    setTemplateResolver(resolver)
}
```

**Comment by cy6erGn0m** [comment]
Thanks!

**Pull request merged**